django/db/models/sql/compiler.py

```diff
@@ -336,9 +336,10 @@ def quote_name_unless_alias(self, name):
         """
         if name in self.quote_cache:
             return self.quote_cache[name]
-        if ((name in self.query.alias_map and name not in self.query.table_map) or
+        tables = [getattr(t, 'table', None) for t in self.query.table_map]
```

Comment:
Nitpick but since `tables` order doesn't matter I would make it a `set` instead. It looks like it only used for `in` lookups.